tests/templates/test_subroutines/test_commuting_evolution.py

```diff
@@ -135,14 +134,37 @@ def test_invalid_hamiltonian(self):
         """Tests TypeError is raised if `hamiltonian` does not have a pauli rep."""
 
         invalid_operator = qml.Hermitian(np.eye(2), 0)
-
         assert pytest.raises(TypeError, qml.CommutingEvolution, invalid_operator, 1)
 
 
 class TestGradients:
     """Tests that correct gradients are obtained for `CommutingEvolution` when frequencies
     are specified."""
 
+    @pytest.mark.unit
+    def test_grad_method_and_recipe(self):
+        """Tests that CommutingEvolution returns the correct grad method and recipe."""
+
+        time = qml.numpy.array(0.1)
+        H = qml.Hamiltonian([0.5, 0.5], [qml.X(0), qml.Y(0)])
+        op = qml.CommutingEvolution(H, time, frequencies=(2,))
+        assert op.grad_method == "A"
+        assert qml.math.allclose(
+            op.grad_recipe[0], [[1.0, 1.0, 0.78539816], [-1.0, 1.0, -0.78539816]]
+        )
+        assert op.grad_recipe[1:] == (None, None)
+
+    @pytest.mark.unit
+    def test_grad_method_is_none_trainable_hamiltonian(self):
+        """Tests that the grad method is None for a trainable Hamiltonian."""
+
+        time = qml.numpy.array(0.1)
+        coeffs = qml.numpy.array([0.5, 0.5])
+        H = qml.Hamiltonian(coeffs, [qml.X(0), qml.Y(0)])
+        op = qml.CommutingEvolution(H, time, frequencies=(2,))
+        assert op.grad_method is None
+        assert op.grad_recipe == [None, None, None]
+

```

Comment:
Mind also adding a test for something like `qml.X(0) + qml.numpy.array(0.5) *qml.Y(0)`?

Reply:
Done